compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -839,4 +844,11 @@ impl<'a, 'tcx> MutVisitor<'tcx> for Integrator<'a, 'tcx> {
     fn visit_source_scope(&mut self, scope: &mut SourceScope) {
         *scope = self.scope_map[*scope];
     }
+
+    fn visit_span(&mut self, span: &mut Span) {
+        *span = self.callsite_span.fresh_expansion(ExpnData {
+            def_site: self.body_span,
+            ..ExpnData::default(ExpnKind::Inlined, *span, self.tcx.sess.edition(), None)
```

Comment:
Should the edition be the edition of `span`? We could be inlining something from another crate which has a different edition than our current one.

Reply:
I was wondering that, too, but I don't see how the edition is relevant from the MIR on anyway, all of that checking happens waay earlier. I can't extract the edition from the `span`, because it may not have any `ExpnData`, so I would have to find the crate of the `Span` and check its edition. Doing something like this from a `Span` is something I don't think we've done before